kernel/src/sched.rs

```diff
@@ -313,6 +317,7 @@ impl Kernel {
                         && !DynamicDeferredCall::global_instance_calls_pending().unwrap_or(false)
                         && self.processes_blocked()
                     {
+                        chip.watchdog().suspend();
                         chip.sleep();
```

Comment:
```suggestion
                        chip.watchdog().suspend();
                        chip.sleep();
                        chip.watchdog.().tickle();
```

I think given the HIL interface as listed, there should be a `tickle` here to make sure the watchdog is active once the chip is active again